pkg/aggregator/internal/limiter/limiter.go

```diff
@@ -47,6 +56,20 @@ func New(limit int, keyTagName string, telemetryTagNames []string) *Limiter {
 		return nil
 	}
 
+	return newLimiter(limit, math.MaxInt, 0, keyTagName, telemetryTagNames)
+}
+
+// NewGlobal returns a limiter with a global per-instance limit, that
+// will be equally distributed between origins.
+func NewGlobal(global int, maxAge int, key string, tags []string) *Limiter {
+	if global <= 0 || global == math.MaxInt {
```

Comment:
Should we consider 

```
	if limit <= 0 {
		return nil
	}
```

Here as well?

Reply:
Not sure I follow. There is no `limit` parameter in this function now, do you mean we should add it? What would it mean?

Reply:
Not sure I follow, there is no `limit` variable in this function now. Do you mean to add an extra parameter?

Reply:
Yep sorry, I definitely got confused here. Then, just to make sure I got it right: it works despite `newLimiter` called with `limit = 0` because global is non-0/non-infinite and `updateLimit` then sets a proper `limit`?

Reply:
Yes, that's right.